### PR TITLE
Add draft documentation for Application Startup

### PR DESCRIPTION
**Pull request by diversemix**
#### Background

Helps developers understand how and where the app is started.

#### Any relevant tickets

Prequel to #1724

#### How has this been tested?

No need.

### CODE REVIEW

**Review by hdrury1 — APPROVED** [review]

File: docs/developing/app-startup.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Dockerfile:CMD []
+Dockerfile-development:CMD []
+```
+
+However, because the `Dockerfile` is meant for developers it does run the application
+explcitly with `RUN [ "npx", "pubsweet", "build"]`
+
+## Docker Compose
+
+In general docker-compose is used as the way to start the application.
```
Comment (will-byrne) [inline docs/developing/app-startup.md]: `In general` repeated as the first two words on the first two sections. This one could be replaced with `docker-compose can be`

Comment (diversemix) [reply docs/developing/app-startup.md]: @will-byrne changed... please :eyes:

**Review by will-byrne — APPROVED** [review]

**Review by Cooryd — APPROVED** [review]

File: docs/developing/app-startup.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Dockerfiles don't specify the way in which the application starts (see below).
+
+```
+~/dev/elife-xpub $ grep CMD Dockerfile*
+Dockerfile:CMD []
+Dockerfile-development:CMD []
+```
+
+However, because the `Dockerfile` is meant for developers it does run the application
```
Comment (giorgiosironi) [inline docs/developing/app-startup.md:13]: That
- doesn't run the application: it is a command executed at image build time
- it is not meant for developers, `Dockerfile-development` is?

File: docs/developing/app-startup.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+76:                sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml run --rm --name elife-xpub_wait_postgres app bash -c './scripts/wait-for-it.sh postgres:5432'"
+```
+
+## Starting in Staging / Prodcution
+
+Below shows the docker compose file used for deploying when `builder` is used.
+
+This is for staging and production:
+
+```
```
Comment (giorgiosironi) [inline docs/developing/app-startup.md:64]: Worth a link since it's another repository?

**Pull request merged**